src/MailCollector.php

```diff
@@ -1476,12 +1476,19 @@ public function getHeaders(\Laminas\Mail\Storage\Message $message)
             $subject = '';
         }
 
+       // secu on message_id setting
+        try {
+            $message_id = $message->getHeader('message_id')->getFieldValue();
+        } catch (Laminas\Mail\Storage\Exception\InvalidArgumentException $e) {
+            $message_id = '';
+        }  
```

Comment:
1. Prefer usage of `getHeaders()->has()`, code is easier to understand.
2. Unless I am wrong, `$message_id` will be used to handle manual import of rejected email, so I guess it is preferable to generate a unique but not random value. 

```suggestion
        $message_id = $message->getHeaders()->has('message-id')
            ? $message->getHeader('message-id')->getFieldValue()
            : 'MISSING_ID_' . sha1($message->getHeaders()->toString());
```

Reply:
Agree